docs/reference/ml/anomaly-detection/ml-configuring-aggregations.asciidoc

```diff
@@ -134,25 +138,135 @@ PUT _ml/datafeeds/datafeed-farequote
+----------------------------------
+PUT _ml/anomaly_detectors/farequote-composite
+{
+  "analysis_config": {
+    "bucket_span": "60m",
+    "detectors": [{
+      "function": "mean",
+      "field_name": "responsetime",
+      "by_field_name": "airline"
+    }],
+    "summary_count_field_name": "doc_count"
+  },
+  "data_description": {
+    "time_field":"time"
+  }
+}
+----------------------------------
+// TEST[skip:setup:farequote_data]
+
+This is an example of a datafeed that uses a `composite` aggregation to bucket
+the metrics based on time and terms:
+
+[source,console]
+----------------------------------
+PUT _ml/datafeeds/datafeed-farequote-composite
+{
+  "job_id": "farequote-composite",
+  "indices": [
+    "farequote"
+  ],
+  "aggregations": {
+    "buckets": {
+      "composite": {
+        "size": 1000,  <1>
+        "sources": [
+          {
+            "time_bucket": {  <2>
+              "date_histogram": {
+                "field": "time",
+                "fixed_interval": "360s",
+                "time_zone": "UTC"
+              }
+            }
+          },
+          {
+            "airline": {  <3>
+              "terms": {
+                "field": "airline"
+              }
+            }
+          }
+        ]
+      },
+      "aggregations": {
+        "time": {  <4>
+          "max": {
+            "field": "time"
+          }
+        },
+        "responsetime": { <5>
+          "avg": {
+            "field": "responsetime"
+          }
+        }
+      }
+    }
+  }
+}
+----------------------------------
+// TEST[skip:setup:farequote_job]
 
+<1> Provide the `size` to the composite agg to control how many resources
+are used when aggregating the data. A larger `size` means a faster datafeed but
+more cluster resources are used when searching.
+<2> The required `date_histogram` composite aggregation source. Make sure it
+is named differently than your desired time field.
+<3> Instead of using a regular `term` aggregation, adding a composite
+aggregation `term` source with the name `airline` works. Note its name
+is the same as the field.
+<4> The required `max` aggregation whose name is the time field in the
+job analysis config.
+<5> The `avg` aggregation is named `responsetime` and its field is also named
+`responsetime`.
 
 [discrete]
 [[aggs-dfeeds]]
 == Nested aggregations in {dfeeds}
 
-{dfeeds-cap} support complex nested aggregations. This example uses the 
-`derivative` pipeline aggregation to find the first order derivative of the 
+{dfeeds-cap} support complex nested aggregations. This example uses the
+`derivative` pipeline aggregation to find the first order derivative of the
 counter `system.network.out.bytes` for each value of the field `beat.name`.
 
+NOTE: `derivative` or other pipeline aggregations may not work within `composite`
+aggregations. See
+{ref}/search-aggregations-bucket-composite-aggregation-pipeline-aggregations.html[composite aggregations and pipeline aggregations].
```

Comment:
@benwtrent Sorry, just realized that `search-aggregations-bucket-composite-aggregation-pipeline-aggregations` is a section on a page called `search-aggregations-bucket-composite-aggregation`. My suggestion below should fix the docs build. Now for real.
```suggestion
{ref}/search-aggregations-bucket-composite-aggregation.html#search-aggregations-bucket-composite-aggregation-pipeline-aggregations[composite aggregations and pipeline aggregations].
```